### DISCUSSION TITLE
主题端有什么方法可以在Thymeleaf的th:if中校验是否已登录吗

### DISCUSSION THREAD

**Discussion by mjsoftking** [halo-dev/halo#7052] [Q&A] [answered]
主题端有什么方法可以在Thymeleaf的th:if中校验是否已登录吗

**Comment by mjsoftking** [comment]
当前需求只需要判断已登录即可

**Comment by guqing** [comment, marked as answer]
@mjsoftking 有以下两种方式可以在主题端判断是否登录
1. 使用 `${#authentication.name}` 取值如果得到 anonymousUser 表示未登录，否则会得到已登录用户的用户名
```html
<p th:if="${#authentication.name != 'anonymousUser'}">
  已登录
</p>
```
2. 使用 [Spring Security Dialect](https://www.thymeleaf.org/doc/articles/springsecurity.html#:~:text=Spring%20Security%20Dialect)
```html
<div sec:authorize="isAuthenticated()">
  已登录
</div>
```

**Comment by mjsoftking** [comment]
@guqing 感谢
